### ISSUE TITLE
CairoMakie saves svg as png

### ISSUE THREAD

**Issue by jfb-h** [JuliaPlots/Makie.jl#1721]
It seems that plots created with CairoMakie aren't correctly saved as svg, making them not editable. See also [this thread](https://discourse.julialang.org/t/makie-svg-saved-with-cairomakie-not-editable-looks-like-raster/75335/11).

Creating a plot via:

```julia
using CairoMakie
p = scatter(rand(5), rand(5))
save("test.svg", p)
```

The source of `test.svg` looks like this:

```svg
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="600pt" height="450pt" viewBox="0 0 600 450" version="1.1">
<defs>
<image id="image107-bHA1gM1y" width="2500" height="1876" xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAACcQAAAdU..."/>
</defs>
<g id="surface1-bHA1gM1y">
<use xlink:href="#image107-bHA1gM1y" transform="matrix(0.24,0,0,0.24,0,0)"/>
</g>
</svg>
```

In line three, it looks like the figure is internally saved as png.

This is on CairoMakie 0.7.4 and Julia 1.7.2.

**Comment by SimonDanisch** [comment]
I can reproduce this...That's pretty weird and definitely not intended.
I don't think anyone switched to this on purpose, and switching to something like this by accident would also be pretty weird. Cairo and Cair_jll haven't been updated in years, so that's also an unlikely the reason.

@jkrumbiegel @ffreyer, we didn't by any chance switch a flag to do this on purpose, did we?

**Comment by jkrumbiegel** [comment]
This happens when the svg engine is supposed to draw something it can't with vector primitives, then it just falls back silently to rasterization. I've noticed this in the past with blending modes, probably we introduced some operation in the stack that triggers this. A git bisect might be the most promising way of finding out as I don't have any idea off the top of my head which change it could have been.
